flow-jrebel-plugin crashes if not run with JRebel

If the dependency `flow-jrebel-plugin` is added to `pom.xml`, but the application is not run with JRebel agent, it crashes (traceback below). It should be enough to show a warning and otherwise do nothing in this case.

```
java.util.ServiceConfigurationError: com.vaadin.flow.server.VaadinServiceInitListener: Provider com.vaadin.flow.server.jrebel.JRebelInitializer could not be instantiated
    at java.util.ServiceLoader.fail (ServiceLoader.java:232)
    at java.util.ServiceLoader.access$100 (ServiceLoader.java:185)
    at java.util.ServiceLoader$LazyIterator.nextService (ServiceLoader.java:384)
    at java.util.ServiceLoader$LazyIterator.next (ServiceLoader.java:404)
    at java.util.ServiceLoader$1.next (ServiceLoader.java:480)
    at java.util.Iterator.forEachRemaining (Iterator.java:116)
    at java.util.Spliterators$IteratorSpliterator.forEachRemaining (Spliterators.java:1801)
    at java.util.stream.ReferencePipeline$Head.forEach (ReferencePipeline.java:580)
    at com.vaadin.flow.server.VaadinService.lambda$init$1 (VaadinService.java:259)
    at com.vaadin.flow.server.VaadinService.runWithServiceContext (VaadinService.java:2298)
    at com.vaadin.flow.server.VaadinService.init (VaadinService.java:257)
    at com.vaadin.flow.server.VaadinServletService.init (VaadinServletService.java:153)
    at com.vaadin.flow.server.VaadinServlet.createServletService (VaadinServlet.java:194)
    at com.vaadin.flow.server.VaadinServlet.createServletService (VaadinServlet.java:175)
    at com.vaadin.flow.server.VaadinServlet.init (VaadinServlet.java:68)
    at org.eclipse.jetty.servlet.ServletHolder.initServlet (ServletHolder.java:671)
    at org.eclipse.jetty.servlet.ServletHolder.getServlet (ServletHolder.java:520)
    at org.eclipse.jetty.servlet.ServletHolder.prepare (ServletHolder.java:814)
    at org.eclipse.jetty.servlet.ServletHandler.doHandle (ServletHandler.java:537)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle (ScopedHandler.java:146)
    at org.eclipse.jetty.security.SecurityHandler.handle (SecurityHandler.java:548)
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle (HandlerWrapper.java:132)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle (ScopedHandler.java:257)
    at org.eclipse.jetty.server.session.SessionHandler.doHandle (SessionHandler.java:1711)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle (ScopedHandler.java:255)
    at org.eclipse.jetty.server.handler.ContextHandler.doHandle (ContextHandler.java:1347)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextScope (ScopedHandler.java:203)
    at org.eclipse.jetty.servlet.ServletHandler.doScope (ServletHandler.java:480)
    at org.eclipse.jetty.server.session.SessionHandler.doScope (SessionHandler.java:1678)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextScope (ScopedHandler.java:201)
    at org.eclipse.jetty.server.handler.ContextHandler.doScope (ContextHandler.java:1249)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle (ScopedHandler.java:144)
    at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle (ContextHandlerCollection.java:220)
    at org.eclipse.jetty.server.handler.HandlerCollection.handle (HandlerCollection.java:152)
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle (HandlerWrapper.java:132)
    at org.eclipse.jetty.server.Server.handle (Server.java:505)
    at org.eclipse.jetty.server.HttpChannel.handle (HttpChannel.java:370)
    at org.eclipse.jetty.server.HttpConnection.onFillable (HttpConnection.java:267)
    at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded (AbstractConnection.java:305)
    at org.eclipse.jetty.io.FillInterest.fillable (FillInterest.java:103)
    at org.eclipse.jetty.io.ChannelEndPoint$2.run (ChannelEndPoint.java:117)
    at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob (QueuedThreadPool.java:781)
    at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run (QueuedThreadPool.java:917)
    at java.lang.Thread.run (Thread.java:748)

Caused by: java.lang.NoClassDefFoundError: org/zeroturnaround/javarebel/ClassEventListener
    at java.lang.Class.getDeclaredConstructors0 (Native Method)
    at java.lang.Class.privateGetDeclaredConstructors (Class.java:2671)
    at java.lang.Class.getConstructor0 (Class.java:3075)
    at java.lang.Class.newInstance (Class.java:412)
    at java.util.ServiceLoader$LazyIterator.nextService (ServiceLoader.java:380)
    at java.util.ServiceLoader$LazyIterator.next (ServiceLoader.java:404)
    at java.util.ServiceLoader$1.next (ServiceLoader.java:480)
    at java.util.Iterator.forEachRemaining (Iterator.java:116)
    at java.util.Spliterators$IteratorSpliterator.forEachRemaining (Spliterators.java:1801)
    at java.util.stream.ReferencePipeline$Head.forEach (ReferencePipeline.java:580)
    at com.vaadin.flow.server.VaadinService.lambda$init$1 (VaadinService.java:259)
    at com.vaadin.flow.server.VaadinService.runWithServiceContext (VaadinService.java:2298)
    at com.vaadin.flow.server.VaadinService.init (VaadinService.java:257)
    at com.vaadin.flow.server.VaadinServletService.init (VaadinServletService.java:153)
    at com.vaadin.flow.server.VaadinServlet.createServletService (VaadinServlet.java:194)
    at com.vaadin.flow.server.VaadinServlet.createServletService (VaadinServlet.java:175)
    at com.vaadin.flow.server.VaadinServlet.init (VaadinServlet.java:68)
    at org.eclipse.jetty.servlet.ServletHolder.initServlet (ServletHolder.java:671)
    at org.eclipse.jetty.servlet.ServletHolder.getServlet (ServletHolder.java:520)
    at org.eclipse.jetty.servlet.ServletHolder.prepare (ServletHolder.java:814)
    at org.eclipse.jetty.servlet.ServletHandler.doHandle (ServletHandler.java:537)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle (ScopedHandler.java:146)
    at org.eclipse.jetty.security.SecurityHandler.handle (SecurityHandler.java:548)
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle (HandlerWrapper.java:132)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle (ScopedHandler.java:257)
    at org.eclipse.jetty.server.session.SessionHandler.doHandle (SessionHandler.java:1711)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle (ScopedHandler.java:255)
    at org.eclipse.jetty.server.handler.ContextHandler.doHandle (ContextHandler.java:1347)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextScope (ScopedHandler.java:203)
    at org.eclipse.jetty.servlet.ServletHandler.doScope (ServletHandler.java:480)
    at org.eclipse.jetty.server.session.SessionHandler.doScope (SessionHandler.java:1678)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextScope (ScopedHandler.java:201)
    at org.eclipse.jetty.server.handler.ContextHandler.doScope (ContextHandler.java:1249)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle (ScopedHandler.java:144)
    at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle (ContextHandlerCollection.java:220)
    at org.eclipse.jetty.server.handler.HandlerCollection.handle (HandlerCollection.java:152)
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle (HandlerWrapper.java:132)
    at org.eclipse.jetty.server.Server.handle (Server.java:505)
    at org.eclipse.jetty.server.HttpChannel.handle (HttpChannel.java:370)
    at org.eclipse.jetty.server.HttpConnection.onFillable (HttpConnection.java:267)
    at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded (AbstractConnection.java:305)
    at org.eclipse.jetty.io.FillInterest.fillable (FillInterest.java:103)
    at org.eclipse.jetty.io.ChannelEndPoint$2.run (ChannelEndPoint.java:117)
    at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob (QueuedThreadPool.java:781)
    at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run (QueuedThreadPool.java:917)
    at java.lang.Thread.run (Thread.java:748)
Caused by: java.lang.ClassNotFoundException: org.zeroturnaround.javarebel.ClassEventListener
    at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass (SelfFirstStrategy.java:50)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass (ClassRealm.java:271)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass (ClassRealm.java:247)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass (ClassRealm.java:239)
    at org.eclipse.jetty.webapp.WebAppClassLoader.loadClass (WebAppClassLoader.java:555)
    at java.lang.ClassLoader.loadClass (ClassLoader.java:357)
    at java.lang.Class.getDeclaredConstructors0 (Native Method)
    at java.lang.Class.privateGetDeclaredConstructors (Class.java:2671)
    at java.lang.Class.getConstructor0 (Class.java:3075)
    at java.lang.Class.newInstance (Class.java:412)
    at java.util.ServiceLoader$LazyIterator.nextService (ServiceLoader.java:380)
    at java.util.ServiceLoader$LazyIterator.next (ServiceLoader.java:404)
    at java.util.ServiceLoader$1.next (ServiceLoader.java:480)
    at java.util.Iterator.forEachRemaining (Iterator.java:116)
    at java.util.Spliterators$IteratorSpliterator.forEachRemaining (Spliterators.java:1801)
    at java.util.stream.ReferencePipeline$Head.forEach (ReferencePipeline.java:580)
    at com.vaadin.flow.server.VaadinService.lambda$init$1 (VaadinService.java:259)
    at com.vaadin.flow.server.VaadinService.runWithServiceContext (VaadinService.java:2298)
    at com.vaadin.flow.server.VaadinService.init (VaadinService.java:257)
    at com.vaadin.flow.server.VaadinServletService.init (VaadinServletService.java:153)
    at com.vaadin.flow.server.VaadinServlet.createServletService (VaadinServlet.java:194)
    at com.vaadin.flow.server.VaadinServlet.createServletService (VaadinServlet.java:175)
    at com.vaadin.flow.server.VaadinServlet.init (VaadinServlet.java:68)
    at org.eclipse.jetty.servlet.ServletHolder.initServlet (ServletHolder.java:671)
    at org.eclipse.jetty.servlet.ServletHolder.getServlet (ServletHolder.java:520)
    at org.eclipse.jetty.servlet.ServletHolder.prepare (ServletHolder.java:814)
    at org.eclipse.jetty.servlet.ServletHandler.doHandle (ServletHandler.java:537)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle (ScopedHandler.java:146)
    at org.eclipse.jetty.security.SecurityHandler.handle (SecurityHandler.java:548)
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle (HandlerWrapper.java:132)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle (ScopedHandler.java:257)
    at org.eclipse.jetty.server.session.SessionHandler.doHandle (SessionHandler.java:1711)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle (ScopedHandler.java:255)
    at org.eclipse.jetty.server.handler.ContextHandler.doHandle (ContextHandler.java:1347)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextScope (ScopedHandler.java:203)
    at org.eclipse.jetty.servlet.ServletHandler.doScope (ServletHandler.java:480)
    at org.eclipse.jetty.server.session.SessionHandler.doScope (SessionHandler.java:1678)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextScope (ScopedHandler.java:201)
    at org.eclipse.jetty.server.handler.ContextHandler.doScope (ContextHandler.java:1249)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle (ScopedHandler.java:144)
    at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle (ContextHandlerCollection.java:220)
    at org.eclipse.jetty.server.handler.HandlerCollection.handle (HandlerCollection.java:152)
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle (HandlerWrapper.java:132)
    at org.eclipse.jetty.server.Server.handle (Server.java:505)
    at org.eclipse.jetty.server.HttpChannel.handle (HttpChannel.java:370)
    at org.eclipse.jetty.server.HttpConnection.onFillable (HttpConnection.java:267)
    at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded (AbstractConnection.java:305)
    at org.eclipse.jetty.io.FillInterest.fillable (FillInterest.java:103)
    at org.eclipse.jetty.io.ChannelEndPoint$2.run (ChannelEndPoint.java:117)
    at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob (QueuedThreadPool.java:781)
    at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run (QueuedThreadPool.java:917)
    at java.lang.Thread.run (Thread.java:748)
```

--- a/flow-jrebel-plugin/src/main/java/com/vaadin/flow/server/jrebel/JRebelClassEventListener.java
+++ b/flow-jrebel-plugin/src/main/java/com/vaadin/flow/server/jrebel/JRebelClassEventListener.java
@@ -1,3 +1,18 @@
+/*
+ * Copyright 2000-2020 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
 package com.vaadin.flow.server.jrebel;
 
 import java.util.concurrent.ScheduledFuture;

--- a/flow-jrebel-plugin/src/main/java/com/vaadin/flow/server/jrebel/JRebelClassEventListener.java
+++ b/flow-jrebel-plugin/src/main/java/com/vaadin/flow/server/jrebel/JRebelClassEventListener.java
@@ -1,0 +1,115 @@
+package com.vaadin.flow.server.jrebel;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.zeroturnaround.javarebel.ClassEventListener;
+import org.zeroturnaround.javarebel.ReloaderFactory;
+import org.zeroturnaround.javarebel.integration.generic.ClassEventListenerAdapter;
+import org.zeroturnaround.javarebel.integration.util.WeakUtil;
+
+import com.vaadin.flow.internal.BrowserLiveReload;
+import com.vaadin.flow.internal.BrowserLiveReloadAccess;
+import com.vaadin.flow.server.VaadinContext;
+import com.vaadin.flow.server.VaadinService;
+
+class JRebelClassEventListener extends ClassEventListenerAdapter {
+
+    static final int RELOAD_DELAY = 100;
+
+    private ScheduledThreadPoolExecutor executor = new ScheduledThreadPoolExecutor(
+            1);
+
+    private ScheduledFuture<?> schedule;
+    private ClassEventRunnable command;
+
+    private Lock lock = new ReentrantLock(true);
+
+    private VaadinService vaadinService;
+
+    /**
+     * Create a new class event listener for JRebel.
+     *
+     * @param vaadinService
+     */
+    public JRebelClassEventListener(VaadinService vaadinService) {
+        super(0);
+
+        this.vaadinService = vaadinService;
+        VaadinContext vaadinContext = vaadinService.getContext();
+        vaadinContext.setAttribute(new JRebelListenerReference(this));
+
+        ReloaderFactory.getInstance()
+                .addClassReloadListener(WeakUtil.weak(this));
+    }
+
+    @Override
+    public void onClassEvent(int eventType, Class<?> klass) {
+        lock.lock();
+
+        try {
+            getLogger().info("JRebel class event with type {}, on class {}",
+                    eventType, klass.getName());
+
+            if (command != null) {
+                command.cancel.set(true);
+                schedule.cancel(false);
+            }
+
+            command = new ClassEventRunnable();
+            schedule = executor.schedule(command, RELOAD_DELAY,
+                    TimeUnit.MILLISECONDS);
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    private class ClassEventRunnable implements Runnable {
+
+        private AtomicBoolean cancel = new AtomicBoolean(false);
+
+        @Override
+        public void run() {
+            lock.lock();
+
+            try {
+                if (cancel.get()) {
+                    return;
+                }
+
+                BrowserLiveReloadAccess liveReloadAccess = vaadinService
+                        .getInstantiator()
+                        .getOrCreate(BrowserLiveReloadAccess.class);
+                BrowserLiveReload liveReload = liveReloadAccess
+                        .getLiveReload(vaadinService);
+                if (liveReload != null) {
+                    liveReload.reload();
+                    getLogger().info("Browser reloaded.");
+                } else {
+                    getLogger().warn("BrowserLiveReload not loaded.");
+                }
+            } finally {
+                lock.unlock();
+            }
+        }
+    }
+
+    private static Logger getLogger() {
+        return LoggerFactory.getLogger(JRebelClassEventListener.class.getName());
+    }
+
+    static class JRebelListenerReference {
+
+        final ClassEventListener listener;
+
+        JRebelListenerReference(ClassEventListener listener) {
+            this.listener = listener;
+        }
+    }
+}

--- a/flow-jrebel-plugin/src/main/java/com/vaadin/flow/server/jrebel/JRebelInitializer.java
+++ b/flow-jrebel-plugin/src/main/java/com/vaadin/flow/server/jrebel/JRebelInitializer.java
@@ -15,25 +15,15 @@
  */
 package com.vaadin.flow.server.jrebel;
 
-import java.util.concurrent.ScheduledFuture;
-import java.util.concurrent.ScheduledThreadPoolExecutor;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicBoolean;
-import java.util.concurrent.locks.Lock;
-import java.util.concurrent.locks.ReentrantLock;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
 
-import com.vaadin.flow.internal.BrowserLiveReload;
-import com.vaadin.flow.internal.BrowserLiveReloadAccess;
 import com.vaadin.flow.server.ServiceInitEvent;
-import com.vaadin.flow.server.VaadinContext;
 import com.vaadin.flow.server.VaadinService;
 import com.vaadin.flow.server.VaadinServiceInitListener;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.zeroturnaround.javarebel.ClassEventListener;
-import org.zeroturnaround.javarebel.ReloaderFactory;
-import org.zeroturnaround.javarebel.integration.generic.ClassEventListenerAdapter;
-import org.zeroturnaround.javarebel.integration.util.WeakUtil;
 
 /**
  * Initialize and maintain JRebel {@link ClassEventListener} which receives
@@ -41,102 +31,32 @@ import org.zeroturnaround.javarebel.integration.util.WeakUtil;
  */
 public class JRebelInitializer implements VaadinServiceInitListener {
 
-    static final int RELOAD_DELAY = 100;
-
     @Override
     public void serviceInit(ServiceInitEvent event) {
-        VaadinService vaadinService = event.getSource();
-
-        ClassEventListener listener = new ClassEventListenerImpl(vaadinService);
-        ReloaderFactory.getInstance()
-                .addClassReloadListener(WeakUtil.weak(listener));
-
-        VaadinContext vaadinContext = vaadinService.getContext();
-        vaadinContext.setAttribute(new JRebelListenerReference(listener));
-
-        getLogger().info("Started JRebel initializer");
-    }
-
-    static class JRebelListenerReference {
-
-        final ClassEventListener listener;
-
-        private JRebelListenerReference(ClassEventListener listener) {
-            this.listener = listener;
-        }
-    }
-
-    private static class ClassEventListenerImpl
-            extends ClassEventListenerAdapter {
-
-        private ScheduledThreadPoolExecutor executor = new ScheduledThreadPoolExecutor(
-                1);
-
-        private ScheduledFuture<?> schedule;
-        private ClassEventRunnable command;
-
-        private Lock lock = new ReentrantLock(true);
-
-        private VaadinService vaadinService;
-
-        private ClassEventListenerImpl(VaadinService vaadinService) {
-            super(0);
-
-            this.vaadinService = vaadinService;
-        }
-
-        @Override
-        public void onClassEvent(int eventType, Class<?> klass) {
-            lock.lock();
-
-            try {
-                getLogger().info(
-                        "JRebel class event with type {}, on class {}",
-                        eventType, klass.getName());
-
-                if (command != null) {
-                    command.cancel.set(true);
-                    schedule.cancel(false);
-                }
-
-                command = new ClassEventRunnable();
-                schedule = executor.schedule(command, RELOAD_DELAY,
-                        TimeUnit.MILLISECONDS);
-            } finally {
-                lock.unlock();
+        final String genericErrorMessage = "Unable to initialize JRebel plugin";
+        // Check if we have JRebel on the classpath
+        try {
+            Class<?> classEventListenerImplClass = Class.forName(
+                    "com.vaadin.flow.server.jrebel.JRebelClassEventListener");
+            Constructor ctor = classEventListenerImplClass
+                    .getConstructor(VaadinService.class);
+            ctor.newInstance(event.getSource());
+            getLogger().info("Started JRebel initializer");
+        } catch (NoClassDefFoundError e) {
+            // Failed because classloader could not load a JRebel class; just
+            // log a warning in this case
+            // (https://github.com/vaadin/flow/issues/7875)
+            if (e.getMessage().contains("org/zeroturnaround/javarebel")) {
+                getLogger().warn(
+                        "Unable to initialize JRebel plugin; ensure that you are running the application under the JRebel agent");
+            } else {
+                getLogger().error(genericErrorMessage, e);
             }
+        } catch (NoSuchMethodException | InstantiationException
+                | IllegalAccessException | InvocationTargetException
+                | ClassNotFoundException e) {
+            getLogger().error(genericErrorMessage, e);
         }
-
-        private class ClassEventRunnable implements Runnable {
-
-            private AtomicBoolean cancel = new AtomicBoolean(false);
-
-            @Override
-            public void run() {
-                lock.lock();
-
-                try {
-                    if (cancel.get()) {
-                        return;
-                    }
-
-                    BrowserLiveReloadAccess liveReloadAccess = vaadinService
-                            .getInstantiator()
-                            .getOrCreate(BrowserLiveReloadAccess.class);
-                    BrowserLiveReload liveReload = liveReloadAccess
-                            .getLiveReload(vaadinService);
-                    if (liveReload != null) {
-                        liveReload.reload();
-                        getLogger().info("Browser reloaded.");
-                    } else {
-                        getLogger().warn("BrowserLiveReload not loaded.");
-                    }
-                } finally {
-                    lock.unlock();
-                }
-            }
-        }
-
     }
 
     private static Logger getLogger() {

--- a/flow-jrebel-plugin/src/test/java/com/vaadin/flow/server/jrebel/JRebelInitializerTest.java
+++ b/flow-jrebel-plugin/src/test/java/com/vaadin/flow/server/jrebel/JRebelInitializerTest.java
@@ -30,7 +30,6 @@ import com.vaadin.flow.internal.BrowserLiveReloadAccess;
 import com.vaadin.flow.server.ServiceInitEvent;
 import com.vaadin.flow.server.VaadinContext;
 import com.vaadin.flow.server.VaadinService;
-import com.vaadin.flow.server.jrebel.JRebelInitializer.JRebelListenerReference;
 
 public class JRebelInitializerTest {
 
@@ -64,7 +63,7 @@ public class JRebelInitializerTest {
         Mockito.when(serviceInitEvent.getSource()).thenReturn(service);
         initializer.serviceInit(serviceInitEvent);
 
-        JRebelListenerReference listenerReference = (JRebelListenerReference) contextAttributeCaptor
+        JRebelClassEventListener.JRebelListenerReference listenerReference = (JRebelClassEventListener.JRebelListenerReference) contextAttributeCaptor
                 .getValue();
         listener = listenerReference.listener;
     }
@@ -76,7 +75,7 @@ public class JRebelInitializerTest {
 
         // Wait for the reload task to be scheduled.
         Mockito.verify(liveReload,
-                Mockito.timeout(JRebelInitializer.RELOAD_DELAY * 3)).reload();
+                Mockito.timeout(JRebelClassEventListener.RELOAD_DELAY * 3)).reload();
     }
 
     private static class JRebelView extends Component {

